File: RepoTransBench/target_projects/Java/C#/Jude95_Utils/public_tests/JActivityManagerPublicTest.cs
using Xunit;
using System.Collections.Generic;
using Jude95_Utils;

namespace Jude95_Utils.PublicTests
{
    public class JActivityManagerPublicTest
    {
        [Fact]
        public void TestStackAdditionAndRetrievalPublic()
        {
            var a1 = new DummyActivity("FirstActivity");
            var a2 = new DummyActivity("SecondActivity");

            var stack = JActivityManager.GetActivityStack();
            stack.Clear();

            Assert.Null(JActivityManager.CurrentActivity());

            var cb = JActivityManager.GetActivityLifecycleCallbacks();
            cb.OnActivityResumed(a1);
            Assert.Equal(a1, JActivityManager.CurrentActivity());

            cb.OnActivityResumed(a2);
            Assert.Equal(a2, JActivityManager.CurrentActivity());

            cb.OnActivityDestroyed(a1);
            Assert.Equal(a2, JActivityManager.CurrentActivity());

            cb.OnActivityDestroyed(a2);
            Assert.Null(JActivityManager.CurrentActivity());
        }

        [Fact]
        public void TestCloseActivityPublic()
        {
            var a1 = new DummyActivity("FirstActivity");
            var a2 = new DummyActivity("SecondActivity");

            var cb = JActivityManager.GetActivityLifecycleCallbacks();
            cb.OnActivityResumed(a1);
            cb.OnActivityResumed(a2);

            JActivityManager.CloseActivity(a1);
            Assert.Equal(a2, JActivityManager.CurrentActivity());

            JActivityManager.CloseActivity(null);
            Assert.Equal(a2, JActivityManager.CurrentActivity());
        }

        [Fact]
        public void TestCloseAllActivityPublic()
        {
            var a1 = new DummyActivity("XActivity");
            var a2 = new DummyActivity("YActivity");
            var cb = JActivityManager.GetActivityLifecycleCallbacks();
            cb.OnActivityResumed(a1);
            cb.OnActivityResumed(a2);

            new JActivityManager().CloseAllActivity();
            Assert.Null(JActivityManager.CurrentActivity());
        }

        [Fact]
        public void TestCloseActivityByNamePublic()
        {
            var a1 = new DummyActivity("TestA");
            var a2 = new DummyActivity("TestB");
            var cb = JActivityManager.GetActivityLifecycleCallbacks();
            cb.OnActivityResumed(a1);
            cb.OnActivityResumed(a2);

            JActivityManager.CloseActivityByName("com.jude.utils.TestA");
            Assert.Equal(a2, JActivityManager.CurrentActivity());
        }

        [Fact]
        public void TestGetCurrentActivityNamePublic()
        {
            var a1 = new DummyActivity("LandingScreen");
            var cb = JActivityManager.GetActivityLifecycleCallbacks();
            cb.OnActivityResumed(a1);

            Assert.Equal("com.jude.utils.LandingScreen", JActivityManager.GetCurrentActivityName());

            cb.OnActivityDestroyed(a1);
            Assert.Equal("", JActivityManager.GetCurrentActivityName());
        }

        [Fact]
        public void TestGetActivityStackPublic()
        {
            var a1 = new DummyActivity("SomeActivity");
            var a2 = new DummyActivity("AnotherActivity");
            var cb = JActivityManager.GetActivityLifecycleCallbacks();
            cb.OnActivityResumed(a1);
            cb.OnActivityResumed(a2);

            var stack1 = JActivityManager.GetActivityStack();
            Assert.False(stack1.Count == 0);
            var stack2 = JActivityManager.GetActivityStack();
            Assert.Equal(stack1, stack2);
        }
    }
}

File: RepoTransBench/target_projects/Java/Go/Jude95_Utils/public_tests/jactivitymanager_public_test.go
package public_tests

import (
	"sync"
	"testing"
)

type DummyActivity struct {
	Name string
}

type activityStackType struct {
	stack []*DummyActivity
	mu    sync.Mutex
}

var activityStack = &activityStackType{}

func (a *activityStackType) Clear() {
	a.mu.Lock()
	defer a.mu.Unlock()
	a.stack = nil
}

func (a *activityStackType) Push(act *DummyActivity) {
	a.mu.Lock()
	defer a.mu.Unlock()
	a.stack = append(a.stack, act)
}

func (a *activityStackType) Remove(act *DummyActivity) {
	a.mu.Lock()
	defer a.mu.Unlock()
	res := []*DummyActivity{}
	for _, e := range a.stack {
		if e != act {
			res = append(res, e)
		}
	}
	a.stack = res
}

func (a *activityStackType) Top() *DummyActivity {
	a.mu.Lock()
	defer a.mu.Unlock()
	if len(a.stack) == 0 {
		return nil
	}
	return a.stack[len(a.stack)-1]
}

func (a *activityStackType) All() []*DummyActivity {
	a.mu.Lock()
	defer a.mu.Unlock()
	return append([]*DummyActivity(nil), a.stack...)
}

func currentActivity() *DummyActivity {
	return activityStack.Top()
}

func closeActivity(act *DummyActivity) {
	if act == nil {
		return
	}
	activityStack.Remove(act)
}
func closeAllActivity() {
	activityStack.Clear()
}
func closeActivityByName(name string) {
	activityStack.mu.Lock()
	defer activityStack.mu.Unlock()
	acts := []*DummyActivity{}
	for _, act := range activityStack.stack {
		if act.Name != name && ("com.jude.utils."+act.Name) != name {
			acts = append(acts, act)
		}
	}
	activityStack.stack = acts
}
func getCurrentActivityName() string {
	act := currentActivity()
	if act == nil {
		return ""
	}
	return "com.jude.utils." + act.Name
}
func getActivityStack() []*DummyActivity {
	return activityStack.All()
}

func TestStackAdditionAndRetrievalPublic(t *testing.T) {
	a1 := &DummyActivity{"FirstActivity"}
	a2 := &DummyActivity{"SecondActivity"}

	activityStack.Clear()
	if currentActivity() != nil {
		t.Errorf("Expected nil")
	}
	activityStack.Push(a1)
	if currentActivity() != a1 {
		t.Errorf("Should be a1")
	}
	activityStack.Push(a2)
	if currentActivity() != a2 {
		t.Errorf("Should be a2")
	}
	closeActivity(a1)
	if currentActivity() != a2 {
		t.Errorf("Should be a2 after closing a1")
	}
	closeActivity(a2)
	if currentActivity() != nil {
		t.Errorf("Should be nil after closing a2")
	}
}

func TestCloseActivityPublic(t *testing.T) {
	a1 := &DummyActivity{"FirstActivity"}
	a2 := &DummyActivity{"SecondActivity"}
	activityStack.Clear()
	activityStack.Push(a1)
	activityStack.Push(a2)
	closeActivity(a1)
	if currentActivity() != a2 {
		t.Errorf("Should be a2")
	}
	closeActivity(nil)
	if currentActivity() != a2 {
		t.Errorf("Should still be a2 after closing nil")
	}
}

func TestCloseAllActivityPublic(t *testing.T) {
	a1 := &DummyActivity{"XActivity"}
	a2 := &DummyActivity{"YActivity"}
	activityStack.Clear()
	activityStack.Push(a1)
	activityStack.Push(a2)
	closeAllActivity()
	if currentActivity() != nil {
		t.Errorf("After closing all, should be nil")
	}
}

func TestCloseActivityByNamePublic(t *testing.T) {
	a1 := &DummyActivity{"TestA"}
	a2 := &DummyActivity{"TestB"}
	activityStack.Clear()
	activityStack.Push(a1)
	activityStack.Push(a2)
	closeActivityByName("com.jude.utils.TestA")
	if currentActivity() != a2 {
		t.Errorf("Should be a2 after close by name TestA")
	}
}

func TestGetCurrentActivityNamePublic(t *testing.T) {
	a1 := &DummyActivity{"LandingScreen"}
	activityStack.Clear()
	activityStack.Push(a1)
	if getCurrentActivityName() != "com.jude.utils.LandingScreen" {
		t.Errorf("Should be LandingScreen")
	}
	closeActivity(a1)
	if getCurrentActivityName() != "" {
		t.Errorf("Should be blank after closing activity")
	}
}

func TestGetActivityStackPublic(t *testing.T) {
	a1 := &DummyActivity{"SomeActivity"}
	a2 := &DummyActivity{"AnotherActivity"}
	activityStack.Clear()
	activityStack.Push(a1)
	activityStack.Push(a2)
	stack1 := getActivityStack()
	if len(stack1) == 0 {
		t.Errorf("Should not be empty")
	}
	stack2 := getActivityStack()
	if len(stack1) != len(stack2) {
		t.Errorf("Stack lengths should match")
	}
}